timedctl/timedctl.py

```diff
@@ -25,708 +24,568 @@
-@click.option("--customer", default=None)
-@click.option("--project", default=None)
-@click.option("--task", default=None)
-@click.option("--description", default=None)
-@click.option("--duration", default=None)
-@click.option("--show-archived", default=False, is_flag=True)
-def add_report(
-    customer,
-    project,
-    task,
-    description,
-    duration,
-    show_archived,
-):  # ruff: noqa: PLR0913
-    """Add report(s)."""
-    # select a task
-    task_id = select_task(customer, project, task, show_archived)
-    # ask the user to enter a description
-    if not description:
-        msg("Enter a description")
-        # get description
-        description = click.prompt("")
-    # ask the user to enter a duration
-    if duration:
-        if re.match(r"^\d{1,2}:\d{2}:\d{2}$", duration) is None:
-            error_handler("ERR_INVALID_DURATION")
-    else:
-        duration = time_picker()
-    # create the report
-    res = timed.reports.post(
-        {"duration": duration, "comment": description},
-        {
-            "task": task_id,
-        },
-    )
-    if res.status_code == requests.codes["created"]:
-        msg("Report created successfully")
-        return
-    # handle exception
-    error_handler("ERR_REPORT_CREATION_FAILED")
-
-
-@add.command("holiday", aliases=["h"])
-def add_holiday():
-    """Add holiday(s)."""
-    error_handler("ERR_NOT_IMPLEMENTED")
-
-
-@add.command("absence", aliases=["abs"])
-def add_absence():
-    """Add absence(s)."""
-    error_handler("ERR_NOT_IMPLEMENTED")
-
-
-@timedctl.group(cls=ClickAliasedGroup, aliases=["e", "edit", "update"])
-def edit():
-    """Edit different things."""
-    # pylint: disable=W0107
-
-
-@edit.command("report", aliases=["r"])
-@click.option("--date", default=None)
-def edit_report(date):
-    """Edit report(s)."""
-    report = select_report(date)
-
-    msg("Comment", True)
-    comment = click.prompt("", default=report[1].strip())
-    duration = time_picker(default=report[2])
-    res = pyfzf.FzfPrompt().prompt(["No", "Yes"], "--prompt 'Are you sure?'")
-    if res == ["Yes"]:
-        res = timed.reports.patch(
-            report[-1],
-            {"comment": comment, "duration": duration},
-            {"task": report[-2]},
+            error_handler("ERR_DELETION_ABORTED")
+
+    def add_report(
+        self,
+        customer,
+        project,
+        task,
+        description,
+        duration,
+        show_archived,
+    ):
+        """Add report(s)."""
+        # select a task
+        task_id = self.select_task(customer, project, task, show_archived)
+        # ask the user to enter a description
+        if not description:
+            msg("Enter a description")
+            # get description
+            description = click.prompt("")
+        # ask the user to enter a duration
+        if duration:
+            if re.match(r"^\d{1,2}:\d{2}:\d{2}$", duration) is None:
+                error_handler("ERR_INVALID_DURATION")
```

Comment:
this can be refactored like this:
```suggestion
        if duration and re.match(r"^\d{1,2}:\d{2}:\d{2}$", duration) is None:
            error_handler("ERR_INVALID_DURATION")
```